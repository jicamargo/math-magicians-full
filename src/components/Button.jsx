// eslint-disable-next-line no-unused-vars
import React from 'react';
import PropTypes from 'prop-types';
import styles from '@/styles/calculator.module.css';
function Button(props) {
  const {
    text, className, onClick,
  } = props;
  return (
    <button 
      type="button"
      className={`${styles.button} ${className}`} 
      onClick={onClick}
    >
      {text}
    </button>
  );
}

Button.defaultProps = {
  className: '',
};

Button.propTypes = {
  text: PropTypes.string.isRequired,
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired,
};

export default Button;
