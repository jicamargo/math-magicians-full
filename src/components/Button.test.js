// eslint-disable-next-line no-unused-vars
import React from 'react';
import { render, screen } from '@testing-library/react';
import Button from './Button';

describe('Button', () => {
  it('renders the Button component', () => {
    render(<Button text="Click Me" onClick={() => {}} />);
    expect(screen.getByRole('button')).toMatchSnapshot();
  });

  it('calls the onClick handler when the button is clicked', () => {
    const onClickMock = jest.fn();
    render(<Button text="Click Me" onClick={onClickMock} />);
    const buttonElement = screen.getByRole('button');
    buttonElement.click();
    expect(onClickMock).toHaveBeenCalled();
  });
});