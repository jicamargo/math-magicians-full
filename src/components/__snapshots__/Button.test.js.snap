// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders the Button component 1`] = `
<button
  class="button "
  type="button"
>
  Click Me
</button>
`;
